### PR TITLE
feat: adds redis auth.

### LINKED ISSUE #1613: Add redis auth
**Description**

Would y'all be open to a PR that enables redis auth? I've never used golang, but it looks fairly straightforward to implement using the [options struct](https://github.com/go-session/redis/blob/4e7f0dcd632c5ada68449f4fedcc806ef66e52ce/options.go#L13).

### PR DESCRIPTION
**Pull request by ianhundere**
#### Summary

This PR enables auth using the [ redis options struct](https://github.com/go-session/redis/blob/4e7f0dcd632c5ada68449f4fedcc806ef66e52ce/options.go#L13) to include a flag for a password/auth.

Closes #1613

#### Release Note

* Adds Redis Authentication

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -161,6 +161,12 @@ func ConfigureAPI(treeID uint) {
 	}
 	if viper.GetBool("enable_retrieve_api") || viper.GetBool("enable_stable_checkpoint") ||
 		slices.Contains(viper.GetStringSlice("enabled_api_endpoints"), "searchIndex") {
+		redisClient = redis.NewClient(&redis.Options{
```
Comment (Hayden-IO) [inline pkg/api/api.go]: This shouldn't be needed, the provider will create the Redis client - https://github.com/sigstore/rekor/blob/main/pkg/indexstorage/indexstorage.go#L34

**Review by Hayden-IO — DISMISSED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (ianhundere) [reply pkg/api/api.go]: ah, thanks for catching that.

**Comment by Hayden-IO** [comment]
@bobcallaway Is the issue here that we need to set up go before CI is run?

**Comment by bobcallaway** [comment]
looks like a `go mod tidy` is needed, as well as an update to `docker-compose.test.yml` to include the password

**Comment by ianhundere** [comment]
> looks like a `go mod tidy` is needed, as well as an update to `docker-compose.test.yml` to include the password

thanks, i ran `go mod tidy` and added a redis pw to `docker-compose.test.yml`.

**Review by Hayden-IO — APPROVED** [review]

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**